datafusion/functions-array/src/udf.rs

```diff
@@ -75,8 +116,233 @@ impl ScalarUDFImpl for ArrayToString {
     }
 
     fn invoke(&self, args: &[ColumnarValue]) -> datafusion_common::Result<ColumnarValue> {
-        let args = ColumnarValue::values_to_arrays(args)?;
-        crate::kernels::array_to_string(&args).map(ColumnarValue::Array)
+        make_scalar_function_with_hints(crate::kernels::array_to_string)(args)
+    }
+
+    fn aliases(&self) -> &[String] {
+        &self.aliases
+    }
+}
+
+#[derive(Debug)]
+pub(super) struct ArrayAppend {
+    signature: Signature,
+    aliases: Vec<String>,
+}
+
+impl ArrayAppend {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::array_and_element(Volatility::Immutable),
+            aliases: vec![
+                String::from("array_append"),
+                String::from("list_append"),
+                String::from("array_push_back"),
+                String::from("list_push_back"),
+            ],
+        }
+    }
+}
+
+impl ScalarUDFImpl for ArrayAppend {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        "array_append"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn return_type(&self, arg_types: &[DataType]) -> datafusion_common::Result<DataType> {
+        Ok(arg_types[0].clone())
+    }
+
+    fn invoke(&self, args: &[ColumnarValue]) -> datafusion_common::Result<ColumnarValue> {
+        make_scalar_function_with_hints(crate::kernels::array_append)(args)
```

Comment:
I think we dont need this. I play around it and find that there is error in 

https://github.com/apache/arrow-datafusion/blob/32d906fc9622af3a67b3828700272092fe0982a0/datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs#L526-L530

LargeList casting panics

If the list casting is fixed, probably we dont need `make_scalar_function_with_hints` anymore.

Reply:
yes, if no `make_scalar_function_with_hints` function, [arrow-datafusion/datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs](https://github.com/apache/arrow-datafusion/blob/32d906fc9622af3a67b3828700272092fe0982a0/datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs#L526-L530) will pannic, cause by arrow-rs not support cast LargeList, So At present, we still need `make_scalar_function_with_hints`

Reply:
we can use as_list::<i64> for large list

Reply:
but I think `make_scalar_function_with_hints` function have clear logic, determine output is Scalar or Array, I want 
 to keep it 🤔

Reply:
Actually I think we will never output Scalar for ArrayFunction, but I'm not 100% sure.

Reply:
I'm not sure if there is any case that we need Scalar here. If so, we definitely can keep `make_scalar_function_with_hints `, otherwise, we can add it back until we need it.

Reply:
maybe @alamb can give some advise 😄